src/UpdateCheck.lua

```diff
@@ -45,14 +45,33 @@ local function downloadFileText(source, file)
 	end
 end
 local function downloadFile(source, file, outName)
-	local text = downloadFileText(source, file)
-	if text then
-		local outFile = io.open(outName, "wb")
-		outFile:write(text)
-		outFile:close()
-	else
-		return true
+	for i = 1, 5 do
+		if i > 1 then
+			ConPrintf("Retrying... (%d of 5)", i)
+		end
+		local easy = curl.easy()
+		local escapedUrl = source..easy:escape(file)
+		easy:setopt_url(escapedUrl)
+		easy:setopt(curl.OPT_ACCEPT_ENCODING, "")
+		if connectionProtocol then
+			easy:setopt(curl.OPT_IPRESOLVE, connectionProtocol)
+		end
+		if proxyURL then
+			easy:setopt(curl.OPT_PROXY, proxyURL)
+		end
+		easy:setopt_writefunction(assert(io.open(outName, "wb+")))
```

Comment:
I didn't realize you could just pass the file handle like this since usually you have to call ":write()" on it afterward.

Reply:
Yup. I just copied what the cURL API documentation suggested verbatim.